src/main/resources/XWiki/Macros/Team.xml

```diff
@@ -0,0 +1,1501 @@
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>(function () {
+  function filter (teamDiv) {
+    const l = this.value.toLowerCase();
+    const users = teamDiv.getElementsByClassName('xwikiteam-user');
+    for (let i = 0, len = users.length; i &lt; len; i++) {
+      const user = users[i];
+      user.style.display = (
+        user.textContent.toLowerCase().includes(l) ||
+        user.dataset.username.toLowerCase().includes(l) ||
+        user.dataset.nameWithoutAccents.includes(l)
+      ) ? "" : "none";
+    }
+  }
+
+  function applyCheckbox(teamDiv, checkbox) {
+    if (checkbox.checked) {
+      teamDiv.classList.remove('xwikiteam-usernames-hidden');
+    } else {
+      teamDiv.classList.add('xwikiteam-usernames-hidden');
+    }
+  }
+
+  function bindShowUsernameCheckbox(teamDiv, checkbox) {
+    applyCheckbox(teamDiv, checkbox);
+    checkbox.onclick = applyCheckbox.bind(null, teamDiv, checkbox);
+  }
+
+  function toggleTools(teamDiv, e) {
+    const tools = teamDiv.querySelector('.xwikiteam-tools');
+    if (tools) {
+      tools.hidden = !tools.hidden;
+      if (!tools.hidden) {
+        const filterInput = tools.querySelector('.xwikiteam-filter');
+        if (filterInput) {
+          filterInput.focus();
+        }
+      }
+    }
+    if (e) {
+      e.preventDefault();
+    }
+  }
+
+  function init() {
+    const teamDivs = document.getElementsByClassName('xwikiteam');
+    for (let i = 0, len = teamDivs.length; i &lt; len; i++) {
+      const teamDiv = teamDivs[i];
+      const a = teamDiv.querySelector('.xwikiteam-tools-link a');
+      if (a) {
+        if (a.onclick) {
+          continue; // don't initialize already initialized team macros
+        }
+        a.onclick = toggleTools.bind(null, teamDiv);
+        a.parentNode.hidden = false;
+      }
+      const ul = teamDiv.querySelector('.xwikiteam-ul');
+      const tools = teamDiv.querySelector('.xwikiteam-tools');
+      if (!tools) continue;
+      const filterInput = tools.querySelector('.xwikiteam-filter');
+      if (filterInput) {
+        filterInput.addEventListener('keyup', filter.bind(filterInput, teamDiv));
+        if (filterInput.value) {
+          filter.call(filterInput, teamDiv);
+          toggleTools(teamDiv)
+        }
+      }
+
+      bindShowUsernameCheckbox(teamDiv, teamDiv.querySelector(".xwikiteam-show-username-checkbox"));
+    }
+  }
+
+  $(document).on('xwiki:dom:updated', init);
```

Comment:
Where is ``$`` defined? You simply copied the code, but ``$`` was actually a parameter of the require block (that you don't have), and was pointing to jQuery. In your code ``$`` currently points to ``Prototype.js``.

Reply:
Whoops, I expected JQuery to be unconditionally available in XWiki. I'll rewrite this in classic javascript.

Reply:
fixed

Reply:
Note that I haven't managed to make this work in the wysiwyg editor, but I don't think the features enabled by this script are relevant in the editor. It's probably better if they are not there when editing.

Reply:
> You simply copied the code

This is extra confusing because the code ran as expected when testing (on page load)